code/modules/events/viral_infection.dm

```diff
@@ -1,66 +1,58 @@
-/var/global/list/event_viruses = list() // so that event viruses are kept around for admin logs, rather than being GCed
-
 datum/event/viral_infection
-	var/list/viruses = list()
-	ic_name = "a biohazard"
-	no_fake = 1//Probability in announce complicates it
+	var/datum/disease2/disease/generated/generated_disease
+	ic_name = "a viral outbreak"
 
 datum/event/viral_infection/setup()
 	announceWhen = rand(0, 3000)
 	endWhen = announceWhen + 1
 
-	//generate 1-3 viruses. This way there's an upper limit on how many individual diseases need to be cured if many people are initially infected
-	var/num_diseases = rand(1,3)
-	for (var/i=0, i < num_diseases, i++)
-		var/datum/disease2/disease/D = new /datum/disease2/disease
+	var/has_virologist = FALSE
+	var/list/valid_diseases = list()
+
+	for(var/mob/living/carbon/human/M in mob_list)
+		if(M.get_assignment() == "Virologist")
```

Comment:
You might want to check if the virologist is active and not afk / ssd / disconnected

Reply:
Good idea. I'll check the virus code for that.

Reply:
I added checks for whether or not the virologist has been AFK and if they're actually currently being played.